src/Components/Blazor/Server/src/Builder/BlazorHostingEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Blazor.Server;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Provides extension methods for hosting client-side Blazor applications in ASP.NET Core.
+    /// </summary>
+    public static class BlazorHostingEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a low-priority endpoint that will serve the the file specified by <paramref name="filePath"/> from the client-side
+        /// Blazor application specified by <typeparamref name="TClientApp"/>.
+        /// </summary>
+        /// <typeparam name="TClientApp">A type in the client-side application.</typeparam>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="filePath">The relative path to the entry point of the client-side application.</param>
```

Comment:
This description might need tweaking. What is the path relative to? I think it's to the `wwwroot` directory within the client-side application - is that right?

Reply:
yes. I'll clarify